application/core/plugins/AuthLDAP/AuthLDAP.php

```diff
@@ -478,9 +477,9 @@ public function newUserSession()
         $groupsearchbase        = $this->get('groupsearchbase');
         $groupsearchfilter      = $this->get('groupsearchfilter');
 
-        // Try to connect
+        /* Get the conexion, createConnection return an error in array, never return false */
         $ldapconn = $this->createConnection();
-        if ($ldapconn === false) {
+        if (isset($ldapconn['errorCode'])) {
```

Comment:
We can check is_array, like you want.

Reply:
> We can check is_array, like you want.

Seems a good idea.

If the connection is OK and if the connection is rejected, the objectes returned may collide a bit?
I mean they are not as different as a scalar and an array as the used to be. Now they are both non-scalar.
Will that create problems?

Reply:
> Seems a good idea.
>
> If the connection is OK and if the connection is rejected, the objectes returned may collide a bit?
> I mean they are not as different as a scalar and an array as the used to be. Now they are both non-scalar.
> Will that create problems?

I do not understand …

in 7.4 : createConnection() return array or ressource
in 8.1 : createConnection() return array or Class

But it return array always with 'errorCode' set, never without (currently)